### PR TITLE
181 remove perspective

### PR DESCRIPTION
**Pull request by serjiokov**
Removed rust perspective definition from plugin.xml
Provided perspective switching to finalPerspective attribute;
Disable TestPespective class;

Signed-off-by: Sergei Kovalchuk <serjiokov@gmail.com>

### CODE REVIEW

File: org.eclipse.corrosion.tests/src/org/eclipse/corrosion/tests/TestPerspective.java [hunk 1]
```diff
@@ -29,8 +29,10 @@
 import org.eclipse.ui.wizards.newresource.BasicNewFileResourceWizard;
 import org.eclipse.ui.wizards.newresource.BasicNewFolderResourceWizard;
 import org.junit.Before;
+import org.junit.Ignore;
 import org.junit.Test;
 
+@Ignore
 public class TestPerspective {
```
Comment (mickaelistria) [inline org.eclipse.corrosion.tests/src/org/eclipse/corrosion/tests/TestPerspective.java]: If it's not relevant anymore, please just remove it.

File: org.eclipse.corrosion/src/org/eclipse/corrosion/wizards/newproject/NewCargoProjectWizard.java [hunk 2]
```diff
@@ -205,6 +216,10 @@ public boolean performFinish() {
 		return true;
 	}
 
+	private void switchPerspective() {
```
Comment (mickaelistria) [inline org.eclipse.corrosion/src/org/eclipse/corrosion/wizards/newproject/NewCargoProjectWizard.java]: I don't think we need a method for a 1-line instruction. Let's just put this instruction directly in the code.

**Comment by serjiokov** [comment]
> The commit message need to have a "Signed-Off-By: ..." footer to conform with Eclipse rules.
> Also, I'm still skeptical whether this is an improvement or not: let's imagine a user in the C++ perspective or another one, and they create a Rust project, why would we bother suggesting them to get to the Resources perspective? I'm still not sure there is a usability gain here.

Mickael Istria, 
You right, that switching dialog sometimes is bothering the user, may remove preferred perspective from plugin.xml and keep the code for further cases? 
But the user needs to switch to custom perspective or standard to use launch bar with Debug and Run.

**Comment by mickaelistria** [comment]
As @norru mentions in the issue, it may be more profitable to investigate addition of Run/Debug icons in the Resources view. Some plugins manage to do it, Corrosion could probably do it too without dealing with a perspective switch.

File: org.eclipse.corrosion.tests/src/org/eclipse/corrosion/tests/TestPerspective.java (hunk 1 above)
Comment (serjiokov) [reply org.eclipse.corrosion.tests/src/org/eclipse/corrosion/tests/TestPerspective.java]: The test was removed.

**Comment by serjiokov** [comment]
> As @norru mentions in the issue, it may be more profitable to investigate addition of Run/Debug icons in the Resources view. Some plugins manage to do it, Corrosion could probably do it too without dealing with a perspective switch.

One more approach to using org.eclipse.launchbar.core the LaunchBar were available from such a perspective like Resource (please see the screen below).

**Comment by serjiokov** [comment]
ice_screenshot_20190117-134423

**Comment by norru** [comment]
This looks nice - I still have concerns about this change though:

https://github.com/eclipse/corrosion/issues/181#issuecomment-455127642

**Comment by norru** [comment]
Ok, I think it's settled that we're removing the perspective.

Would it be a great deal for @serjiokov to also add this to the Resource perspective?

selection_125

**Comment by serjiokov** [comment]
Sorry was overloaded on the last week,let me please summarise:
- contribution  of rust perspective was removed;
- activity to add launch bar moved to another thread #220
 Current status #220 - on progress;

**Comment by mickaelistria** [comment]
Replaced with https://github.com/eclipse/corrosion/commit/94fe2935f0cf405835376ac0709077b6d9c14085

**Pull request closed without merging**